src/inmanta/ast/statements/define.py

```diff
@@ -148,14 +163,20 @@ def evaluate(self) -> None:
 
             add_attributes: Dict[str, Attribute] = {}
             for attribute in self.attributes:
-                attr_type = self.namespace.get_type(attribute.type)
+                attr_type: Type = attribute.type.get_type(self.namespace)
                 if not isinstance(attr_type, (Type, type)):
                     raise TypingException(self, "Attributes can only be a type. Entities need to be defined as relations.")
 
                 name = str(attribute.name)
-                attr_obj = Attribute(entity_type, attr_type, name, attribute.multi, attribute.nullable)
+                attr_obj = Attribute(
```

Comment:
perhaps attribute can be similarly be cleaned up by passing a Type?

Reply:
I agree, but the type is also used for ResultVariable creation so it wouldn't be a straightforward replacement. I'll make a seperate ticket for this.

Reply:
see #1770